feat(south-modbus): Modbus address offset (ON TOP OF #1128)

Lulu300 opened the pull request:
This feature allows to define an offset on the addresses of the points to be requested in order to be compatible with the JBus protocol. When using JBus, all the addresses are offset by -1, the addition of this new field avoids having to increment the addresses of all the points manually.

burgerni10 commented on src/south/Modbus/Modbus.schema.jsx:
```diff
@@ -53,6 +53,15 @@ schema.form = {
+    type: 'OIbSelect',
+    md: 2,
+    newRow: false,
+    options: [0, -1],
```
I don't know much about the use cases for the offset, but if it's only for Modbus / Jbus, the select should display these options instead of the integer. On the backend, depending of the value (Modbus or JBus) the right offset (0 or -1) is used. Default setting would be of course Modbus (instead of 0). What do you think ?

Lulu300 replied:
I think it'll be only Modbus and JBus, you're right I'll make the changes. I also have a question, is it possible in the options field to provide key value instead of only value (Like HTML5 Select Form) ?

burgerni10 replied:
Right now, it is not possible : the options are an array of values directly used as it is by the OibSelect component. That's why a small process in the class file may be used to convert JBus to "-1" and Modbus to "0"

burgerni10 reviewed (commented):
I rebased your branch so it is up to date with the master branch. I updated your commits to reflect the conventional commit format we use. See https://www.conventionalcommits.org/en/v1.0.0/ for more info. It helps us to automate our release changelogs.

Lulu300 commented:
> 
> 
> I rebased your branch so it is up to date with the master branch. I updated your commits to reflect the conventional commit format we use. See https://www.conventionalcommits.org/en/v1.0.0/ for more info. It helps us to automate our release changelogs.

I'll be careful to respect this convention for future commits

burgerni10 commented:
Hi @Lulu300, thank you for your work, your PR seems fine now :)

Before accepting it, I was wondering about JBus / Modbus : from what I understand, JBus  addresses from 0 instead of 1 for Modbus, hence the offset.

My question is : does a user directly get the JBus address from the PLC (in this case the offset is confusing because it will be the correct address right away) or does a JBus PLC give the Modbus address to the user and in this case the offset make sense ?

Lulu300 commented:
In fact it depends on the PLC manufacturer but usually they provide the user with the Modbus address, hence we need the offset to make the configuration in OIBus look similar to what's inside the PLC documentation.

burgerni10 commented:
Ok, thanks !
